docs/node-mixin/alerts/alerts.libsonnet

```diff
@@ -184,6 +184,39 @@
               description: '{{ $labels.instance }} interface {{ $labels.device }} has encountered {{ printf "%.0f" $value }} transmit errors in the last two minutes.',
             },
           },
+          {
+            alert: 'NodeClockSkewDetected',
+            expr: |||
+              (
+                instance:node_timex_offset:abs > 0.05
+              and
+                deriv(instance:node_timex_offset:abs[2m]) >= 0
```

Comment:
You could use sub-queries to avoid the recording rule.

But perhaps the `abs` way is not even mathematically correct.

How about the following:
```PromQL
(
  (
    instance:node_timex_offset:abs > 0.05
  and
    deriv(instance:node_timex_offset:abs[5m]) >= 0
  )
or
  (
    instance:node_timex_offset:abs < 0.05
  and
    deriv(instance:node_timex_offset:abs[5m]) <= 0
  )
)
```

I also increased the range a bit to get less noise.

Reply:
I have tested the proposed query and it is firing almost all the time as following is true on a healthy node:
```
  (
    instance:node_timex_offset:abs < 0.05
  and
    deriv(instance:node_timex_offset:abs[5m]) <= 0
  )
```

---
As for math, I checked it experimentally and it works. This is what I get on my test bench:

Raw series:
![raw](https://user-images.githubusercontent.com/3531758/64948367-32699980-d877-11e9-9674-5599e27267ab.png)

Deriv other raw series:
![deriv](https://user-images.githubusercontent.com/3531758/64948402-457c6980-d877-11e9-9c0f-614e8edfed1c.png)

Deriv over abs:
![deriv_abs](https://user-images.githubusercontent.com/3531758/64948406-4ad9b400-d877-11e9-96fa-bbd51dcac69f.png)

When looking at light green series it was clear to me that series needs to be normalized with `abs()` before calculating derivative. I might be wrong here, but tests show this is working as expected.

Reply:
Sorry, I mixed things up in two ways. For one, I used the rule name when I wanted to use the metric name. And then the `for` clause needs to be longer than the range for the `deriv`.

In your test bench, how does the following alerting expression work, with a `for` clause of 10m?

```PromQL
(
  (
    node_timex_offset_seconds > 0.05
  and
    deriv(node_timex_offset_seconds[5m]) >= 0
  )
or
  (
    node_timex_offset_seconds < 0.05
  and
    deriv(node_timex_offset_seconds[5m]) <= 0
  )
)
```

Reply:
I didn't have any problems with such expression and I changed to using this one :)